### PR TITLE
[ci] Move _testing.py into tests folder

### PR DESCRIPTION
**Pull request by ailzhang**
Related issue = #3782

### CODE REVIEW

**Review by rexwangcc — APPROVED** [review]
LGTM

**Review by k-ye — APPROVED** [review]
LGTM!

File: tests/python/bls_test_template.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import numpy as np
 
 import taichi as ti
+from tests import test_utils
```
Comment (strongoier) [inline tests/python/bls_test_template.py:6]: It seems like this file doesn't use `test_utils`.

File: tests/python/fuse_test_template.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import time
 
 import taichi as ti
+from tests import test_utils
```
Comment (strongoier) [inline tests/python/fuse_test_template.py:4]: ditto

File: tests/python/test_ad_atomic.py [hunk 3]
```diff
@@ -1,9 +1,9 @@
-from taichi._testing import approx
-
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import approx
```
Comment (strongoier) [inline tests/python/test_ad_atomic.py:3]: Can we use `test_utils.approx` in this file?

File: tests/python/test_ad_basics.py [hunk 4]
```diff
@@ -2,9 +2,10 @@
 
 import numpy as np
 import pytest
-from taichi._testing import approx
 
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import approx
```
Comment (strongoier) [inline tests/python/test_ad_basics.py:8]: ditto

File: tests/python/test_ast_refactor.py [hunk 5]
```diff
@@ -1,13 +1,14 @@
 import numpy as np
 import pytest
-from taichi._testing import approx
 from taichi.lang import impl
 from taichi.lang.util import has_pytorch
 
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import approx
```
Comment (strongoier) [inline tests/python/test_ast_refactor.py:8]: ditto

File: tests/python/test_atomic.py [hunk 6]
```diff
@@ -1,6 +1,6 @@
-from taichi._testing import approx
-
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import approx
```
Comment (strongoier) [inline tests/python/test_atomic.py:3]: ditto

File: tests/python/test_bit_operations.py [hunk 7]
```diff
@@ -2,12 +2,13 @@
 
 import numpy as np
 import pytest
-from taichi._testing import allclose
 
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import allclose
```
Comment (strongoier) [inline tests/python/test_bit_operations.py:8]: ditto

File: tests/python/test_element_wise.py [hunk 8]
```diff
@@ -1,8 +1,9 @@
 import numpy as np
 import pytest
-from taichi._testing import allclose
 
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import allclose
```
Comment (strongoier) [inline tests/python/test_element_wise.py:6]: ditto

File: tests/python/test_f16.py [hunk 9]
```diff
@@ -2,15 +2,16 @@
 
 import numpy as np
 import pytest
-from taichi._testing import approx
 from taichi.lang.util import has_pytorch
 
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import approx
```
Comment (strongoier) [inline tests/python/test_f16.py:9]: ditto

File: tests/python/test_gui.py [hunk 10]
```diff
@@ -1,13 +1,14 @@
 import numpy as np
 import pytest
-from taichi._testing import make_temp_file
 from taichi.lang.misc import get_host_arch_list
 
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import make_temp_file
```
Comment (strongoier) [inline tests/python/test_gui.py:7]: ditto

File: tests/python/test_image_io.py [hunk 11]
```diff
@@ -2,11 +2,12 @@
 
 import numpy as np
 import pytest
-from taichi._testing import make_temp_file
 from taichi.lang.misc import get_host_arch_list
 from taichi.lang.util import to_numpy_type
 
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import make_temp_file
```
Comment (strongoier) [inline tests/python/test_image_io.py:10]: ditto

File: tests/python/test_linalg.py [hunk 12]
```diff
@@ -2,13 +2,14 @@
 
 import numpy as np
 import pytest
-from taichi._testing import approx
 from taichi.lang.misc import get_host_arch_list
 
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import approx
```
Comment (strongoier) [inline tests/python/test_linalg.py:9]: ditto

File: tests/python/test_matrix.py [hunk 13]
```diff
@@ -3,11 +3,12 @@
 
 import numpy as np
 import pytest
-from taichi._testing import approx
 from taichi.lang import impl
 from taichi.lang.misc import get_host_arch_list
 
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import approx
```
Comment (strongoier) [inline tests/python/test_matrix.py:11]: ditto

File: tests/python/test_mpm88.py [hunk 14]
```diff
@@ -1,9 +1,10 @@
 import os
 
 import pytest
-from taichi._testing import approx
 
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import approx
```
Comment (strongoier) [inline tests/python/test_mpm88.py:7]: ditto

File: tests/python/test_random.py [hunk 15]
```diff
@@ -1,9 +1,9 @@
-from taichi._testing import approx
-
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import approx
```
Comment (strongoier) [inline tests/python/test_random.py:3]: ditto

File: tests/python/test_return.py [hunk 16]
```diff
@@ -1,10 +1,11 @@
 import pytest
-from taichi._testing import approx
 
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import approx
```
Comment (strongoier) [inline tests/python/test_return.py:5]: ditto

File: tests/python/test_scalar_op.py [hunk 17]
```diff
@@ -2,9 +2,10 @@
 
 import numpy as np
 import pytest
-from taichi._testing import allclose
 
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import allclose
```
Comment (strongoier) [inline tests/python/test_scalar_op.py:8]: ditto

File: tests/python/test_ssa.py [hunk 18]
```diff
@@ -6,12 +6,13 @@
 import math
 
 import numpy as np
-from taichi._testing import approx
 
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import approx
```
Comment (strongoier) [inline tests/python/test_ssa.py:12]: ditto

File: tests/python/test_svd.py [hunk 19]
```diff
@@ -1,10 +1,11 @@
 import numpy as np
-from taichi._testing import approx
 
 import taichi as ti
+from tests import test_utils
+from tests.test_utils import approx
```
Comment (strongoier) [inline tests/python/test_svd.py:5]: ditto

File: tests/python/test_ast_resolver.py [hunk 20]
```diff
@@ -3,6 +3,8 @@
 
 from taichi.lang.ast.symbol_resolver import ASTResolver
 
+from tests import test_utils
```
Comment (strongoier) [inline tests/python/test_ast_resolver.py:6]: This file doesn't use `test_utils`.

File: tests/python/test_binding.py [hunk 21]
```diff
@@ -1,4 +1,5 @@
 import taichi as ti
+from tests import test_utils
```
Comment (strongoier) [inline tests/python/test_binding.py:2]: ditto

File: tests/python/test_cli.py [hunk 22]
```diff
@@ -9,6 +9,7 @@
 from taichi.main import TaichiMain
 
 import taichi as ti
+from tests import test_utils
```
Comment (strongoier) [inline tests/python/test_cli.py:12]: ditto

**Review by strongoier — APPROVED** [review]
LGTM! Some nits to make the code cleaner...

**Comment by ailzhang** [comment]
@strongoier nice catch about the template files (I added the `import` statement to all python files programmatically lol 
 we should name them differently to highlight the fact that they're not test files. I'll do that in a separate PR!

**Pull request merged**